### PR TITLE
users: Add override button to confirm weak password

### PR DESCRIPTION
**Pull request by tomasmatus**
Fixes: #16612

### CODE REVIEW

**Comment by tomasmatus** [comment]
Screenshots

create_weak_password
set_weak_password

File: pkg/users/password-dialogs.js [hunk 1]
```diff
@@ -246,10 +244,33 @@ export function set_password_dialog(account, current_user) {
...
+                                if (change_self) {
+                                    return passwd_self(state.password_old, state.password);
+                                } else {
+                                    return passwd_change(account.name, state.password);
+                                }
+                            } else {
+                                update();
+                                return Promise.reject();
+                            }
+                        });
```
Comment (marusak) [inline pkg/users/password-dialogs.js:269]: This is the same as from line 236. Could be factored out into function.

File: pkg/users/account-create-dialog.js [hunk 2]
```diff
@@ -256,20 +255,39 @@ export function account_create_dialog(accounts) {
...
+                    style: "warning",
+                    clicked: () => {
+                        return validate(state.confirm_weak).then(valid => {
+                            if (valid)
+                                return create();
+                            else {
+                                update();
+                                return Promise.reject();
+                            }
+                        });
```
Comment (marusak) [inline pkg/users/account-create-dialog.js:286]: This is the same as on line 260, could be factored into a function

**Comment by marusak** [comment]
Almost all tests aborted due to some networking outage. I rebased and force pushed to get them a clean run

**Review by marusak — APPROVED** [review]
nice, thank you!

**Comment by marusak** [comment]
Bad news @tomasmatus - this is in conflict now. Can you please rebase? Feel free to ping me and I can help you with it!

File: test/verify/check-users [hunk 3]
```diff
@@ -164,17 +164,17 @@ class TestAccounts(MachineCase):
 
         # Password is weak, lets change it to another weak - this should still not accept
         b.wait_in_text("#accounts-create-dialog .pf-c-form__helper-text.pf-m-error", "Password quality check failed:")
-        b.wait_in_text("#accounts-create-dialog .pf-c-form__helper-text.pf-m-error", "Click Create again to use the password anyway.")
```
Comment (marusak) [inline test/verify/check-users:167]: Oh, can you please replace these with checks that the new button was shown? It is equivalent to that.

File: test/verify/check-users [hunk 4]
```diff
@@ -248,7 +248,6 @@ class TestAccounts(MachineCase):
         b.wait_visible("#account-set-password-meter.danger")
         b.click('#account-set-password-dialog button.apply')
         b.wait_in_text("#account-set-password-dialog .pf-c-form__helper-text.pf-m-error", "Password quality check failed:")
-        b.wait_in_text("#account-set-password-dialog .pf-c-form__helper-text.pf-m-error", "Click Set password again to use the password anyway.")
```
Comment (marusak) [inline test/verify/check-users:251]: Here as well please.

**Review by marusak — CHANGES_REQUESTED** [review]
Can you please add two more checks into tests to make sure the logic is kept the same?

File: test/verify/check-users [hunk 5]
```diff
@@ -164,17 +164,17 @@ class TestAccounts(MachineCase):
 
         # Password is weak, lets change it to another weak - this should still not accept
         b.wait_in_text("#accounts-create-dialog .pf-c-form__helper-text.pf-m-error", "Password quality check failed:")
-        b.wait_in_text("#accounts-create-dialog .pf-c-form__helper-text.pf-m-error", "Click Create again to use the password anyway.")
         b.set_input_text('#accounts-create-password-pw1', "bar")
         b.set_input_text('#accounts-create-password-pw2', "bar")
         b.wait_not_present("#accounts-create-dialog .pf-c-form__helper-text.pf-m-error")
```
Comment (marusak) [inline test/verify/check-users:169]: Plus here then you should wait for the button to disappear.

File: test/verify/check-users (hunk 4 above)
Comment (marusak) [reply test/verify/check-users:251]: And on line 257 you could wait for the button to not appear when `passwords do not match`.

**Comment by tomasmatus** [comment]
Also added checks in a couple more places where the button should not appear.

**Review by marusak — APPROVED** [review]
Nice! Thank you :)

**Pull request merged**